src/site/content/en/blog/window-controls-overlay/index.md

```diff
@@ -0,0 +1,432 @@
+
+```json
+{
+  "display_override": ["window-controls-overlay"]
+}
+```
+
+The window controls overlay will be visible only when all of the following conditions are satisfied:
+
+1. The app is _not_ opened in the browser, but in a separate PWA window.
+1. The manifest includes `"display_override": ["window-controls-overlay"]`. (Other values are
+   allowed thereafter.)
+1. The PWA is running on a desktop operating system.
+1. The current origin matches the origin for which the PWA was installed.
+
+The result of this is an empty title bar area with the regular window controls on the left or the
+right, depending on the operating system.
+
+{% Img src="image/8WbTDNrhLsU0El80frMBGE4eMCD3/7slf2HkHYGyhhLzdBz9Q.png", alt="App window with an empty titlebar with the window controls on the right.", width="800", height="112" %}
+
+### Moving content into the title bar
+
+Now that there is space in the title bar, you can move something there. For this article, I have
+built a Chuck Norris jokes PWA. A useful feature for this app may be a search for words in jokes.
+Fun fact: Chuck Norris has installed this PWA on his iPhone and has let me know he loves the push
+notifications he receives whenever a new joke is submitted. The HTML for the search feature looks
+like this:
+
+```html
+<div class="search">
+  <img src="chuck-norris.png" alt="Chuck Norris" width="32" height="32" />
+  <label>
+    <input type="search" />
+    Search words in jokes
+  </label>
+</div>
+```
+
+To move this `div` up into the title bar, some CSS is needed:
+
+```css
+.search {
+  /* Make sure the `div` stays there, even when scrolling. */
+  position: fixed;
+  /**
+   * Gradient, because why not. Endless opportunities.
+   * The gradient ends in maroon, which happens to be the app's
+   * `<meta name="theme-color" content="maroon">`.
+   */
+  background-image: linear-gradient(90deg, #131313, 33%, maroon);
+  /* Use the environment variable for the left anchoring with a fallback. */
+  left: env(titlebar-area-x, 0);
+  /* Use the environment variable for the top anchoring with a fallback. */
+  top: env(titlebar-area-y, 0);
+  /* Use the environment variable for setting the width with a fallback. */
+  width: env(titlebar-area-width, 100%);
+  /* Use the environment variable for setting the height with a fallback. */
+  height: env(titlebar-area-height, 33px);
+}
+```
+
+You can see the effect of this code in the screenshot below. The title bar is fully responsive. When
+you resize the PWA window, the title bar reacts as if it were composed of regular HTML content,
+which, in fact, it is.
+
+{% Img src="image/8WbTDNrhLsU0El80frMBGE4eMCD3/5dc2j3CfrKczvTaASvKE.png", alt="App window with a search bar in the title bar.", width="800", height="112" %}
+
+### Determining which parts of the title bar are draggable
+
+While the screenshot above suggests that you are done, you are not done quite yet. The PWA window is
```

Comment:
Although it hasn't yet been implemented, there will be a portion of the overlay that is draggable. This protects the user's ability to drag the window in case the app fails to implement a draggable region. My interpretation of this sentence is that the entire overlay is not draggable, which isn't quite true--although _most_ of the overlay will not be draggable.

Reply:
Gotcha. I tried every pixel, but makes sense it didn't work since it isn't implemented yet.